lib/grammers-client/src/client/files.rs

```diff
@@ -231,35 +232,44 @@ impl Client {
 
         if big_file {
             let parts = Arc::new(parts);
-            let mut tasks = Vec::with_capacity(WORKER_COUNT);
-            for _ in 0..WORKER_COUNT {
-                let handle = self.clone();
+            let mut connections = Vec::with_capacity(CONNECTION_COUNT);
+            for _ in 0..CONNECTION_COUNT {
                 let parts = Arc::clone(&parts);
-                let task = async move {
-                    while let Some((part, bytes)) = parts.next_part().await? {
-                        let ok = handle
-                            .invoke(&tl::functions::upload::SaveBigFilePart {
-                                file_id,
-                                file_part: part,
-                                file_total_parts: total_parts,
-                                bytes,
-                            })
-                            .await
-                            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-
-                        if !ok {
-                            return Err(io::Error::new(
-                                io::ErrorKind::Other,
-                                "server failed to store uploaded data",
-                            ));
-                        }
+                let connection = async move {
+                    let mut tasks = Vec::with_capacity(WORKER_COUNT);
+                    let client = Client::connect(self.0.config.clone()).await.unwrap();
+                    for _ in 0..WORKER_COUNT {
+                        let handle = client.clone();
+                        let parts = Arc::clone(&parts);
+                        let task = async move {
+                            while let Some((part, bytes)) = parts.next_part().await? {
+                                let ok = handle
+                                    .invoke(&tl::functions::upload::SaveBigFilePart {
+                                        file_id,
+                                        file_part: part,
+                                        file_total_parts: total_parts,
+                                        bytes,
+                                    })
+                                    .await
+                                    .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+
+                                if !ok {
+                                    return Err(io::Error::new(
+                                        io::ErrorKind::Other,
+                                        "server failed to store uploaded data",
+                                    ));
+                                }
+                            }
+                            Ok(())
+                        };
+                        tasks.push(task);
```

Comment:
I think we should call https://core.telegram.org/mtproto/service_messages#request-to-destroy-session before dropping the extra clients?

When creating a client, deeper down, a random session ID is generated and used. We may just need to add a method on the `mtsender` to invoke that.